### ISSUE TITLE
EULA

### ISSUE THREAD

**Issue by Ataman** [cseelhoff/RimThreaded#23]
This repository is breaching the Rimworld EULA pretty hard. Are you sure Ludeon Studios is going to be okay with that?

For reference: https://rimworldgame.com/eula/

**Comment by cseelhoff** [comment]
yikes. I sure hope I didn't breach any EULA. You mean because so much of the methods contain decompiled vanilla code? Any idea who I should contact to get someone over there to bless off on it? I was able to message Tynan about the mod development, but I did not mention anything as detailed about how I was doing this. Angering him would be the last thing that I would want to do

**Comment by cseelhoff** [comment]
I emailed ludeonhelp@gmail.com and asked them to review. We will see what they say.

**Comment by Ataman** [comment]
> You mean because so much of the methods contain decompiled vanilla code?

Yes. I never had to contact Tynan before but I guess ludeonhelp@gmail.com is a start. A PM to Tynan on the Ludeon Forums might work as well.

**Comment by KiameV** [comment]
Any mod that uses Harmony requires decompiling the code. I do not see this mod in conflict with the Mod section of the EULA

**Comment by Ataman** [comment]
Nobody said that decompiling is the issue.

The issue is decompiling half the game logic, uploading it publicly to Github and then claiming an MIT License on top of it for good measure.

Tynan said he's fine with decompiling and looking at his code, he never said you're allowed to copy more than a few lines from it though.

To patch with Harmony you never need to copy more than a few lines of code if any. That's what Harmony is made for in the first place; to allow code modification without copying any of it. 

But hey, I'm not Tynan and he might be very well okay with it. Which is why you should ask him BEFORE a DMCA request strikes you out of the blue after you spent dozen of hours working on this.

Off topic: You should consider transpiling anyway if you don't want another incompatibility hell like CE.

**Comment by cseelhoff** [comment]
From: Ludeon Support <ludeonhelp@gmail.com>
Date: Fri, 11 Sep 2020 17:35:14 +1200
Message-ID: <CAJ-V48JQi7m+6ugBLG3vsmMSrjPwJRG+MAi-_LKDWWzNqFGtfQ@mail.gmail.com>
Subject: Re: Am I violating the EULA?
To: Caleb Seelhoff <cseelhoff@gmail.com>
Content-Type: multipart/alternative; boundary="0000000000008caf7c05af030c1c"

--0000000000008caf7c05af030c1c
Content-Type: text/plain; charset="UTF-8"

Hello,

I've checked with Tynan and he agrees with my thoughts. My thoughts being
that this mod is perfectly fine and does not break the EULA.
I see it's on the Steam Workshop. I've just subbed to it to give it a go
myself later on. :)

Kind Regards,
Michael
